tests/test_impedance.py

```diff
@@ -0,0 +1,235 @@
+                                                channel_phase=[1, 2, 3.3], channel_linestyle=100.1)
+
+
+def test_save_load():
+    """Unit test for save and load."""
+    # assumption: the given scope object is valid
+    example = pss.Impedance.generate_impedance_object([1, 2, 3], [4, 5, 6],
+                                                      channel_phase=[10, 20, 30],
+                                                      channel_unit="A", channel_label="label", channel_color="red",
+                                                      channel_source="source", channel_linestyle='--')
+
+    # save + load: working example
+    pss.Impedance.save(example, "test_example")
+    loaded_example = pss.Impedance.load("test_example.pkl")
+    assert (example.channel_frequency == loaded_example.channel_frequency).all()
+    assert (example.channel_impedance == loaded_example.channel_impedance).all()
+    assert (example.channel_phase == loaded_example.channel_phase).all()
+    assert example.channel_unit == loaded_example.channel_unit
+    assert example.channel_label == loaded_example.channel_label
+    assert example.channel_color == loaded_example.channel_color
+    assert example.channel_source == loaded_example.channel_source
+    assert example.channel_linestyle == loaded_example.channel_linestyle
+
+    # save: wrong file path type
+    with pytest.raises(TypeError):
+        pss.Impedance.save(example, 123)
+
+    # save: insert wrong scope type
+    with pytest.raises(TypeError):
+        pss.Impedance.save(123, "test_example")
+
+    # load: not a pkl-file
+    with pytest.raises(ValueError):
+        pss.Impedance.load("test_example.m")
+
+    # load: not a string filepath
+    with pytest.raises(TypeError):
+        pss.Impedance.load(123)
+
+    # load: non-existing pkl-file
+    with pytest.raises(ValueError):
+        pss.Impedance.load("test_example_not_existing.pkl")
+
+def test_copy():
+    """Unit test for copy()."""
+    # wrong input type
+    with pytest.raises(TypeError):
+        pss.Impedance.copy("not-a-channel")
+
+    # test for valid copy
+    object_to_copy = pss.Impedance.generate_impedance_object([1, 2, 3], [4, 5, 6], [7, 8, 9],
+                                                             channel_unit="A", channel_label="label", channel_color="red",
+                                                             channel_source="source", channel_linestyle='--')
+    object_copy = pss.Impedance.copy(object_to_copy)
+
+    assert object_copy == object_copy
+
+def test_eq():
+    """Test __eq__()."""
+    ch_1 = pss.Impedance.generate_impedance_object([1, 2, 3], [4, 5, 6], [7, 8, 9],
+                                                   channel_unit="A", channel_label="label", channel_color="red",
+                                                   channel_source="source", channel_linestyle='--')
+    ch_2 = pss.Impedance.copy(ch_1)
```

Comment:
Similar approach also here. Use different types of data and numbers (Positiv and negative integer and float values) are stored and reload correctly.
E.g. [0.1,1,1.2,2.1,3], [0,5.1,1.2,2.1,3], phase: [-4,-5.9,0,6,1.2]
Assumption: Frequency and impedance values shall be only positive, phase values are float or int?

Reply:
done